docs/baseband/tutorials/getting_started.rst

```diff
@@ -256,25 +254,71 @@ VDIF file's headers are of class::
-were only interested in decoding specific threads, we can select them using the
-``thread_ids`` keyword::
+By default, ``fh.read()`` returns complete samples, i.e. with all
+available threads, polarizations or channels. If we were only interested in
+decoding specific components of the complete sample, we can select them by
+passing indexing objects the ``subset`` keyword in open.  For example, if we
+only wanted thread 3 of the sample VDIF file::
 
-    >>> fh = vdif.open(SAMPLE_VDIF, 'rs', thread_ids=[2, 3])
+    >>> fh = vdif.open(SAMPLE_VDIF, 'rs', subset=3, squeeze=False)
     >>> fh.sample_shape
-    SampleShape(nthread=2)
+    SampleShape(nthread=1, nchan=1)
     >>> d = fh.read(20000)
     >>> d.shape
-    (20000, 2)
+    (20000, 1, 1)
+    >>> fh.subset
+    (slice(3, 4, None),)
+    >>> fh.close()
+
+Since ``squeeze=False``, ``subset`` is converted from ``3`` to ``slice(3, 4,
+None)`` to retain dimensions of length unity.  This behaviour is turned off
+when ``squeeze=True`` (see below).  Like ``squeeze``, ``subset`` cannot be
+changed for an already-open file reader.
+
+Data with multi-dimensional samples can be subset by passing a `tuple` of
+indexing objects with the same dimensional ordering as the sample shape prior
+to squeezing (in the case of VDIF this is threads, then channels).  For
+example, if we wished to select threads 1 and 3, and channel 0::
+
+    >>> fh = vdif.open(SAMPLE_VDIF, 'rs', subset=([1, 3], 0), squeeze=False)
+    >>> fh.sample_shape
+    SampleShape(nthread=2, nchan=1)
     >>> fh.close()
 
-For VDIF, ``thread_ids`` selects the specified **threads** (each of which may
-have multiple channels), while for others this selects the specified
-**channels**.
+If a `tuple` is not used when subsetting multi-dimensional data, ``subset``
+will only act upon the the first dimension::
+
+    >>> fh = vdif.open(SAMPLE_VDIF, 'rs', subset=[1, 3], squeeze=False)
+    >>> fh.sample_shape
+    SampleShape(nthread=2, nchan=1)
+    >>> fh.close()
+
+Here, we have also selected 1 and 3, and channel 0.  No enclosing `tuple` is
+required for data with single-dimensional samples.
+
+If ``squeeze=True``, dimensions of length unity are removed from the decoded
+data after subsetting::
+
+    >>> fh = vdif.open(SAMPLE_VDIF, 'rs', subset=([1, 3], 0))
+    >>> fh.sample_shape
+    SampleShape(nthread=2)
+    >>> fh.close()
 
+``subset`` accepts any object `that can be used to for basic indexing
+<https://docs.scipy.org/doc/numpy-1.13.0/reference/arrays.indexing.html>`_ of a
+`numpy.ndarray`, with the exceptions of ``None``.  Advanced indexing (as done
+above with ``subset=([1, 3], 0)``) is also possible, but any indexing that
+leads to a change in the number of dimensions will result in errors.  We can
+use broadcasting to select specific threads from more than one sample shape
+dimension; see the Numpy documentation on `integer array indexing.
+<https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html#integer-array-indexing>`_
+
```

Comment:
This is maybe the place to mention that, for the general case, things will be slightly faster with `subset=slice(1, 4, 2)` instead of `subset=[1, 3]` (unfortunately, this is not true for this specific example, though... so maybe just raise an issue to remind us).